synapse/handlers/auth.py

```diff
@@ -240,11 +239,7 @@ def _check_password_auth(self, authdict, _):
         if not user_id.startswith('@'):
             user_id = UserID.create(user_id, self.hs.hostname).to_string()
 
-        if not (yield self._check_password(user_id, password)):
-            logger.warn("Failed password login for user %s", user_id)
-            raise LoginError(403, "", errcode=Codes.FORBIDDEN)
```

Comment:
No, _check_password is changing from returning `bool` to `raise`, so this code becomes redundant